libraries/lib-mixer/SampleTrackStuff.h

```diff
@@ -0,0 +1,39 @@
+#pragma once
+
+#include "SampleCount.h"
+
+#include <chrono>
+#include <fstream>
+#include <memory>
+#include <vector>
+
+namespace LibSampleTrack
+{
+using BlockPtr = std::shared_ptr<std::vector<float>>;
+using BlockWPtr = std::weak_ptr<std::vector<float>>;
+
+struct MIXER_API ClipBlockSequence
+{
+   // adjacent sample blocks
+   std::vector<BlockPtr> blocks;
+
+   // FillBufferFromTrackBlockSequence needs these two. We keep them here for
+   // convenience.
+   size_t offsetIntoFirstBlock;
+   sampleCount usefulLength;
```

Comment:
I had to read Sequence.cpp to understand that `usefulLength` is a span of sample positions not necessarily covered by the blocks.  Better that I could have read a comment explaining that here.